win64.mak

```diff
@@ -108,7 +108,8 @@ test_hash:
 	"$(DMD)" -m$(MODEL) -conf= -Isrc -defaultlib=$(DRUNTIME) -run test\hash\src\test_hash.d
 
 test_stdcpp:
-	"$(MAKE)" -f test\stdcpp\win64.mak "DMD=$(DMD)" MODEL=$(MODEL) "VCDIR=$(VCDIR)" DRUNTIMELIB=$(DRUNTIME) "CC=$(CC)" test
+	setmscver.bat
+	"$(MAKE)" -f test\stdcpp\win64.mak "DMD=$(DMD)" MODEL=$(MODEL) "VCDIR=$(VCDIR)" DRUNTIMELIB=$(DRUNTIME) "CC=$(CC)" "_MSC_VER=$(_MSC_VER)" test
```

Comment:
This is the CI issue:
`setmscver.bat` sets the `_MSC_VER` variable, but then when make executes the next line (make), it seems like the variable is forgotten...

Reply:
It's not forgotten. You simply execute different processes that fork from the parent environment. Try e.g. `&&`

Reply:
child processes should inherit the parent environment?

Reply:
I can't find an incantation that works... how are you supposed to do this? I need to set a make variable from a thing that a script determines?

Reply:
`setmscver.bat && "$(MAKE)" "_MSC_VER=%_MSC_VER%"` should work.

Reply:
it doesn't... i tried every conceivable combination of stuff like that I could think of.
I found a solution though.

But there's another problem; I just realised that windows make isn't actually make... It's an incomplete make clone!

It won't let me do `ifeq` in the makefile... what's that about? How do I set a variable conditionally?

Reply:
It's Digital Mars make, a very basic make with some surprising behavior. (see https://dlang.org/blog/2018/06/11/dasbetterc-converting-make-c-to-d/)

> How do I set a variable conditionally?

I don't think that is possible directly. Depending on the problem invoking make recursively might be part of a solution.